pkg/chunk/chunk_test.go

```diff
@@ -282,3 +302,76 @@ func benchmarkDecode(b *testing.B, batchSize int) {
 		}
 	}
 }
+
+func TestChunk_Slice(t *testing.T) {
+	now := model.Now()
+
+	// dummy chunk is created with time range now-1hour to now
+	chunkStartTime := now.Add(-time.Hour)
```

Comment:
I personally find these tests a bit hard to read due to the usage of `now`. Have you considered the following:
1. Here you calculate `chunkStartTime` and `chunkEndTime` (based on the number of samples of the chunk and their interval)
2. In the test cases below you never use `now` but always use `chunkStartTime` and `chunkEndTime` to define the time ranges

I believe this way tests would be more clear and less error prone. What's your take?